### PR TITLE
 [RCA Repair Item] Retried deletes should ignore 404 errors

### PR DESCRIPTION
**Pull request by pranavmalik-msft**
**Description:**
This pull request aims to resolve the reported issue related to Azcopy, where it erroneously reports failures in cases of a 404 error, despite a successful background delete operation. The proposed fix involves disregarding 404 errors and treating them as successful operations.

**Changes:**
- Implemented a comprehensive fix to address the observed behavior, ensuring Azcopy operates correctly even when a 404 error is reported.
- Added validation checks to enhance the overall reliability of Azcopy.

### CODE REVIEW

**Review by vibhansa-msft — DISMISSED** [review]

File: ste/xfer-deleteBlobFS.go [hunk 1]
```diff
@@ -51,7 +54,16 @@ func doDeleteHNSResource(jptm IJobPartTransferMgr) {
 	transferDone := func(err error) {
 		status := common.ETransferStatus.Success()
 		if err != nil {
-			status = common.ETransferStatus.Failed()
+			var respErr *azcore.ResponseError
```
Comment (vibhansa-msft) [inline ste/xfer-deleteBlobFS.go:57]: We need to add test cases for this and validate that 404 are ignored while others are reported.

**Review by seanmcc-msft — CHANGES_REQUESTED** [review]
Please add tests.

File: ste/xfer-deleteBlobFS.go (hunk 1 above)
Comment (gapra-msft) [inline ste/xfer-deleteBlobFS.go:57]: Do we only want to apply the ignore 404 to dfs removes? what about file and blob removes?

Comment (seanmcc-msft) [reply ste/xfer-deleteBlobFS.go:57]: Agreed with @gapra-msft, we should be consistent.

Comment (gapra-msft) [reply ste/xfer-deleteBlobFS.go:57]: It looks like for blob and file, we also check 403, should we do a similar thing here?

Comment (pranavmalik-msft) [reply ste/xfer-deleteBlobFS.go:57]: Done in the latest commit

**Pull request closed without merging**